### DISCUSSION TITLE
What's the BaseEntity?

### DISCUSSION THREAD

**Discussion by danielniccoli** [mikro-orm/mikro-orm#5392] [Q&A]
I've been reading about entities in https://mikro-orm.io/docs/defining-entities.

The first example already introduces a `CustomBaseEntity` with no explanation as to what that is.

```
export interface IBook extends CustomBaseEntity {
  title: string;
  author: Author;
  publisher?: Ref<Publisher>;
  tags: Collection<BookTag>;
}

export const Book = new EntitySchema<IBook, CustomBaseEntity>({
  name: 'Book',
  extends: 'CustomBaseEntity',
  properties: {
    title: { type: 'string' },
    author: { kind: 'm:1', entity: 'Author' },
    publisher: { kind: 'm:1', entity: 'Publisher', ref: true, nullable: true },
    tags: { kind: 'm:n', entity: 'BookTag', fixedOrder: true },
  },
});
```

It continues with a short sentence about using custom base entities, but still without any explanation on what that is: https://mikro-orm.io/docs/defining-entities#using-custom-base-entity

Same with https://mikro-orm.io/docs/guide/first-entity and https://mikro-orm.io/docs/guide/relationships.

What is the BaseEntity? Do I need it?

**Comment by B4nan** [comment]
Does this page help?

https://mikro-orm.io/docs/inheritance-mapping#mapped-superclasses
